### ISSUE TITLE
IndexRoute and Link tests

### ISSUE THREAD

**Issue by karantin2020** [rackt/react-router#2297]
Hello, did you see try tests?

```js
describe('Link navigation tests', function () {
    class App extends Component {
      render() {
        return <div>App {this.props.children}</div>
      }
    }
    class MainLayout extends Component {
      render() {
        return (
        <div>
          MainLayout {this.props.children}
          <nav role="navigation" >
            <ul >
              <li>
                <Link id="home" to="/" aria-label="Home" activeClassName="is-active">Home</Link>
              </li>
              <li>
                <Link id="posts" to={"/posts"} activeClassName="is-active">Posts</Link>
              </li>
            </ul>
          </nav>
        </div>
        )
      }
    }
    class Home extends Component {
      render() {
        return <div>Home {this.props.children}</div>
      }
    }
    class Posts extends Component {
      render() {
        return <div>Posts {this.props.children}</div>
      }
    }
    class NotFound extends Component {
      render() {
        return <div>NotFound {this.props.children}</div>
      }
    }
    it("Link to='/' is active when route go to '/posts'", function (done) {

      render((
        <Router history={createHistory('/posts')}>  
          <Route component={App} >
            <Route path="/" component={MainLayout} >
              <IndexRoute component={Home} />
              <Route path="/posts" component={Posts} />
            </Route>
            <Route path="*" component={NotFound} />
          </Route>
        </Router>
      ), node, function () {
        expect(this.history.isActive('/')).toBe(true)
        expect(this.history.isActive('/posts')).toBe(true)
        done()
      })
    })
    it("Link to='/posts' is inactive when route go to '/'", function (done) {

      render((
        <Router history={createHistory('/')}>  
          <Route component={App} >
            <Route path="/" component={MainLayout} >
              <IndexRoute component={Home} />
              <Route path="/posts" component={Posts} />
            </Route>
            <Route path="*" component={NotFound} />
          </Route>
        </Router>
      ), node, function () {
        expect(this.history.isActive('/')).toBe(true)
        expect(this.history.isActive('/posts')).toBe(false)
        done()
      })
    })
    it("IndexLink to='/' is inactive when route go to '/'", function (done) {
      class MainLayout extends Component {
        render() {
          return (
          <div>
            MainLayout {this.props.children}
            <nav role="navigation" >
              <ul >
                <li>
                  <IndexLink id="indexhome" to="/" aria-label="Home" activeClassName="is-active">Home</IndexLink>
                </li>
                <li>
                  <Link id="posts" to={"/posts"} activeClassName="is-active">Posts</Link>
                </li>
              </ul>
            </nav>
          </div>
          )
        }
      }
      render((
        <Router history={createHistory('/')}>  
          <Route component={App} >
            <Route path="/" component={MainLayout} >
              <IndexRoute component={Home} />
              <Route path="/posts" component={Posts} />
            </Route>
            <Route path="*" component={NotFound} />
          </Route>
        </Router>
      ), node, function () {
        expect(this.history.isActive('/')).toBe(true)
        expect(this.history.isActive('/posts')).toBe(false)
        const a = node.querySelector('#indexhome')
        expect(a.className.trim()).toEqual('')
        done()
      })
    })
  })
```

I opened issue about IndexRoute work. It does not work as expected.
In my issue there were code changes.
See https://github.com/rackt/react-router/issues/2231
Thanks

**Comment by taion** [comment]
I don't understand what you're trying to say. Can you please describe what you mean?

**Comment by chikhpykh** [comment]
Why Link to="/" is always active when it is inside parent of indexRoute? If change Link to IndexLink to="/" it is always inactive. Sample is in issue 2231.
When isActive function does path compare then Link to="/" works well.
It was fine in rc1 but was broken in rc3

**Comment by taion** [comment]
`Link` being active is by design. `IndexLink` not being active is most likely the bug fixed by #2237.

**Comment by chikhpykh** [comment]
Active and inactive I mean link status when user clicks links in nav menu, change className and activeClassName
